File: scripting/engine/Playground/ecs/ECSFilter.swift
public struct ECSFilter {
    public enum Usage: Int16 {
        case read = 3
        case write = 4
        case readWrite = 5
    }

    public enum Operation: Int16 {
        case and = 0
        case or = 1
        case not = 2
    }

    internal let id: UInt64
    internal let usage: Int16
    internal let operation: Int16

    public init<T>(component: T.Type, usage: Usage, operation: Operation) {
        self.id = ECSHandler.id(for: T.self)
        self.usage = usage.rawValue
        self.operation = operation.rawValue
    }
}

File: engine/managed/PlaygroundAssembly/ECS/Core/EcsFilter.cs
namespace PlaygroundAssembly.ECS.Core;

public struct EcsFilter
{
    public enum Usage: short
    {
        Read = 3,
        Write = 4,
        ReadWrite = 5
    }

    public enum Operation : short
    {
        And = 0,
        Or = 1,
        Not = 2
    }

    public ulong Id;
    public Usage ParamUsage;
    public Operation ParamOperation;
}
